server/src/server.ts

```diff
@@ -350,12 +350,22 @@ CallStack (from HasCallStack):
 		const charPos = errorJson.ce_position[1] - 1;
 		const suggestions = errorJson.ce_suggestions;
 		const actualMessage = errorJson.ce_errorMessage;
+		const offendingToken = errorJson.ce_offendingToken;
 
-		const start = { line: linePos, character:charPos };
+		const start ={ line: linePos, character: charPos };
+		const end = offendingToken ?
+				{ line: linePos, character: charPos + offendingToken.length }
+			:	{ line: linePos + 1, character: 0 };
+
+		// App options currently have error position at the `:` of a json field: `<k> : <v>`.
+		if (actualMessage.includes('not a valid app option')) {
+			start.character -= offendingToken.length;
+			end.character   -= offendingToken.length;
```

Comment:
Could this cause `end.character` to be negative if end was `{ line: linePos + 1, character: 0 }`?

Reply:
No, since offending tokens will always be present for `not a valid app option` errors. The end position will be always be at least `charPos + offendingToken.length` long